wheel/src/run_generator.rs

```diff
@@ -47,15 +94,77 @@ pub struct PySpendBundleConditions {
     pub cost: u64,
 }
 
+#[pymethods]
+impl PySpendBundleConditions {
+    #[new]
+    fn new(
+        spends: Vec<PySpend>,
+        reserve_fee: u64,
+        height_absolute: u32,
+        seconds_absolute: u64,
+        agg_sig_unsafe: Vec<(Bytes48, Bytes)>,
+        cost: u64,
+    ) -> PySpendBundleConditions {
+        PySpendBundleConditions {
+            spends,
+            reserve_fee,
+            height_absolute,
+            seconds_absolute,
+            agg_sig_unsafe,
+            cost,
+        }
+    }
+
+    #[staticmethod]
+    pub fn from_bytes(blob: &[u8]) -> PyResult<Self> {
+        py_from_bytes::<Self>(blob)
+    }
+
+    // returns the type as well as the number of bytes read from the buffer
+    #[staticmethod]
+    pub fn parse_rust(blob: &[u8]) -> PyResult<(Self, u32)> {
```

Comment:
one problem I have here is that this won't accept a `memoryview` object. So on the python side, I have to force the `memoryview` to `bytes` which isn't great. Any ideas?

Reply:
My understanding is the `memoryview` api is not available when we use the `abi` bindings, which allow us to create just a single wheel that works across python 3.7/8/9/10. If we go to version-specific wheels, then we get `memoryview` back (I guess that API isn't guaranteed to remain the same across versions). So it's a big trade-off in build complexity vs performance. My guess is copying bytes one extra time won't be too much of a performance hit as long as the users of the rust wheel understand that accessing rust fields is expensive so don't do it repeatedly.

Reply:
By the way, since you're moving both serialization and deserialization from python to rust, you don't actually need the format to be backward compatible. You could really use straight `bincode` here rather than the weird `chia_bincode`. [Of course, we'll need the `chia_bincode` to parse blockchain structures that can't change, so maybe we may as well use it everywhere.]

Reply:
> Another option is to allow an optional "cursor" argument, defaulting to 0, so you can use the same bytes objects across multiple parse calls

Good point. I haven't spent too much time thinking of clever ways to integrate this with the existing `Streamable` framework. Right now I just pull out the buffer from `BytesIO` and then advance its cursor when rust is done parsing. I'm not sure it's worth the trouble to try to pass something more stream-like down, as that would be quite complicated. Especially since we only ever use it with `BytesIO` anyway.

I think copying the bytes each time is fine for now. At least it doesn't show up in the profiles. Once everything else is faster it might and we could address it then. (e.g. `FullBlock` objects may be large enough for it to matter).

> By the way, since you're moving both serialization and deserialization from python to rust, you don't actually need the format to be backward compatible

Some of the other expensive objects to serialize and deserialize that I'm hoping to address down the road do need to be backwards compatible. like `FullBlock` and `CoinRecord`.